engine/orchestration/src/main/java/com/cloud/agent/manager/AgentManagerImpl.java

```diff
@@ -596,6 +596,7 @@ protected AgentAttache notifyMonitorsOfConnection(final AgentAttache attache, fi
 
         final Long dcId = host.getDataCenterId();
         final ReadyCommand ready = new ReadyCommand(dcId, host.getId(), NumbersUtil.enableHumanReadableSizes);
+        ready.setWait(60);
```

Comment:
These wait/timeouts should be externalized.

Reply:
I remember somebody complained there are too much global settings.
what's the benefit of adding another setting ? @GutoVeronezi 

normally ReadyCommand is executed in less than 1 second. 
60 seconds is very safe value.

Reply:
In general, hard coded "magic" numbers are not interesting; we bind the system to what we think is proper and do not allow the operators to shape the system according to their use case.

For the timeout/wait settings, we could think on a flexible mecanism to allow operators overriding the timeout/wait values in a granular way, e.g. defining a timeout/wait for each command, and fallbacking to the global setting `wait` in the absence of the specifc setting.

Reply:
@GutoVeronezi 
the problem is actually caused by the falling back to the default `wait` which is 30 mins by default.

Reply:
I agree with @GutoVeronezi , but in this case if it fixes we can add a new ticket for externalizing it and merge this.

Reply:
I am with @weizhouapache on this. We already have a lot of global settings which makes it confusing for the end user as well.

Reply:
@vishesh92 @DaanHoogland @GutoVeronezi @weizhouapache I'm inclined to include this fix in 4.19.0. I'm creating an improvement ticket for follow-up on externalizing the wait. Maybe we can look into some generic solution while addressing it.

Reply:
https://github.com/apache/cloudstack/issues/8506

Reply:
the number of global setting we already have should not be an argument. Nobody know them by head and anybody will have to use search facilities to manage those anyway. We have not the value 60 in three separate extra places for a value that has a default of 1800 (?). Both tuning and code maintenance are served by creating an externalisation for those.
@weizhouapache I understand your argument against the need for tuning, however we are now setting 60 for a value that you say should be one. That sounds like we will want to tune it to 10 or 5 or maybe to 7 afterwards.

Anyway, this is merged. Let's discuss on #8506.